docs/tutorials-redeliver-messages.md

```diff
@@ -0,0 +1,236 @@
+
+* **Cumulative acknowledgment**
+* **Individual acknowledgment**
+
+Cumulative acknowledgment receives a message or a message id as a parameter and marks the messages before the message as
+consumed for this subscription. For multiple-partition topics, the cumulative acknowledgment will work for the single
+partition without impacting other partitions. Individual acknowledgment receives a message or a message id as a parameter
+and only marks this message as consumed for this subscription.
+![img_1.png](../static/img/blog-consume-best-practice/acknowledgement-types.png)
+
+### Messages Redeliver Mechanism
+
+There might be instances where the received messages cannot be processed at this time or some errors happened during processing.
+The client needs to redeliver the unacknowledged messages or a particular message after a delay or immediately.
+Pulsar provides at-least-once semantics when the client does not enable transaction because the client may cache some
+messages out of Pulsar when redelivering messages.
+
+Pulsar Consumer API provides four ways to reconsume the unacknowledged messages later:
+
+* **ackTimeout**
+* **deadLetterPolicy**
+  * **reconsumeLaterCumulative**
+  * **reconsumeLater**
+* **negativeAcknowledge**
+* **redeliverUnacknowledgedMessages**
+
+The **ackTimeout** is an automatic protection mechanism. If a consumer configured ackTimeout, the messages will be
+auto-redelivered when the received messages are not acknowledged after a long time. It works at the client side, ensuring
+the unacknowledged messages will be redelivered to another consumer when the connection of a consumer remains active
+but the business system gets stuck or misses to ack some messages. If the consumer gets disconnected when the client crashes,
+the messages will be auto-redelivered by the broker too. However, this mechanism does not wait for the response of acknowledgment,
+so if an acknowledgment fails on the broker side or proxy side, an ack hole may occur.
+
+The **deadLetterPolicy** is a policy in the message queue used to handle messages that cannot be processed properly.
+In many message queue systems (such as RabbitMQ, Google Pub/Sub, Apache Pulsar, etc.), this strategy is implemented.
+In Pulsar, the deadLetterPolicy is implemented at the client side, it creates a new retry letter topic and dead letter
+topic when building a consumer with `deadLetterPolicy` configuration. When a consumer calls `reconsumeLaterCumulative`
+or `reconsumeLater`, the message (method parameter) will be produced to the retry letter topic until the retry time reaches
+the `maxRedeliverCount`. The message will be produced to the dead letter topic when the retry time reaches the `
+maxRedeliverCount`. The main difference between them is that `reconsumeLaterCumulative` will cumulative ack the
+message (method parameter) after it is produced and `reconsumeLater` will individual ack the message (method parameter)
+after it is produced.
```

Comment:
Let's clarify: 
1. The consumers automatically subscribe to the retry letter topic.
2. It's important that the retry-letter topic name will include both the topic name and the subscription name, since we want to avoid having two subscription reading from the topic, writing their retries into the same retry topic. The retries is a state only in the context of your subscription.